src/Drupal/Composer/DrupalInstallerPlugin.php

```diff
@@ -603,29 +603,56 @@ protected function getPackageName(PackageEvent $event) {
     protected function getPackageVersion(PackageInterface $package) {
         $version = $package->getPrettyVersion();
 
+        // Detect endpoint.
+        $platform = FALSE;
+        $repo = $package->getRepository();
+        if (method_exists($repo, 'getRepoConfig')) {
+            $repo_config = $repo->getRepoConfig();
+            if (isset($repo_config['url'])) {
+                // The new packages.drupal.org separates the platform from the version entirely.
+                $repourl = parse_url($repo_config['url']);
+                if ($repourl['host'] == 'packages.drupal.org') {
+                    $platform = substr($repourl['path'], 1);
```

Comment:
Why is this substr(..., 1).

Reply:
just wanted to grab the stuff after the slash.

Reply:
I was drawing a blank on a less stupid method of grabbing the platform version out of the url.